'use strict';(function(){const indexCfg=;indexCfg.doc={id:'id',field:['title','content'],store:['title','href'],};const index=FlexSearch.create('balance',indexCfg);window.bookSearchIndex=index;index.add({'id':0,'href':'/mfeprimer-3.1/','title':"MFEprimer-3.1",'content':" MFEprimer-3.1 Big change from v3.0\nMajor changes from v3.0  merge several tools into one single execute file mfeprimer support major platforms: Linux, Mac and Windows bug fixed and faster than v3.0 fix thermodynamics calculation bugs for dimers and hairpins local web server version is available  Summary of different versions Before getting started, I\u0026rsquo;d like to make a summary for different versions. Please choose the right version for you.\n   Title web server by the author command-line local web server     primer number \u0026lt;= 50 no limit usually \u0026lt;= 50   database pre-indexed by the author no limit no limit   user interface graphic and easy to use terminal and requires essential Linux skills graphic and easy to use   users from worldwide the owner usually all members of the institution   workflow no yes and can be integrated into user defined primer design workflows no    1. Web servers maintained by the author‌  https://mfeprimer3.igenetech.com/ [version 3.0] web server funded by iGeneTech, a company aims to provide kit and service for target capture and sequencing, also is the company the author works for. http://124.207.243.56:8081/ [version 3.1, for test] web server funded by Prof. Dongsheng Zhao (AMMS, 军事医学科学院). Currently, there is only human genome (hg19) available and will add other genomes later.  2. Command-line version Install  Please download the right version for your machine: https://github.com/quwubin/MFEprimer-3.0/releases (take Linux version as an example). Uncompress the file. Make sure the file has the execute permission. Run with option \u0026ldquo;-h\u0026rdquo; to get help message. You can rename the file to mfeprimer as you wish.\n# download the executable mfeprimer file wget -c https://github.com/quwubin/MFEprimer-3.0/releases/download/v3.1.0/mfeprimer-3.1.0-linux-amd64.gz # uncompress the file gzip -d mfeprimer-3.1.0-linux-amd64.gz # make it executable chmod +x mfeprimer-3.1.0-linux-amd64 # test it ./mfeprimer-3.1.0-linux-amd64 -h MFEprimer-3.1.0, a functional primer quality control program for checking non-specific amplicons, dimers, hairpins and others. Web site: https://www.mfeprimer.com/ Cite: Wang, K., Li, H., Xu, Y., Shao, Q., Yi, J., Wang, R., … Qu, W. (2019). MFEprimer-3.0: quality control for PCR primers. Nucleic Acids Research. https://doi.org/10.1093/nar/gkz351. Usage: mfeprimer [flags] mfeprimer [command] Available Commands: dimer analysis primer dimers hairpin analysis primer hairpins help Help about any command index index fasta file for mfeprimer indexcmp Compare the old and the new index file, return false when they are not same spec analysis specificity of primers version print version number Flags: -b, --bind print specific and nonspecific binding sites and patterns for each primer -d, --db strings [*] database indexed for specificity check: -d hg19.fa -d mrna.fa --diva float concentration of divalent cations [mM] (default 1.5) --dntp float concentration of dNTPs [mM] (default 0.25) -h, --help help for mfeprimer -i, --in string [*] a file contains primer sequences in fasta format -j, --json output in json format -k, --kvalue int k value for format, defaut is 9 (default 9) -S, --maxSize int max product size for the predicted amplicons [bp] (default 2000) -s, --minSize int min product size for the predicted amplicons [bp] --misEnd int mis-match ends to the position of 3' end, 9 (kvalue) for the 9th base from the 3'end (default 9) --misMatch int max allowed mismatches between kmer and its binding sites --misStart int mis-match starts from the position of 3' end, 1 is for the very end of primer. (default 1) --mono float concentration of monovalent cations [mM] (default 50) --oligo float concentration of annealing oligos [nM] (default 50) -o, --out string output file name, e.g., primer.mfe.txt --snp string SNP file name in bed format, e.g., dbsnp151.bed -t, --tm float tm cutoff value to filter the amplicons (default 30) Use \u0026quot;mfeprimer [command] --help\u0026quot; for more information about a command.    In v3.1.0, I make index,dimer,hairpin as sub-commands to mfeprimer, each sub-command has their own arguments and options to set.\nI manually tested the Mac and Linux version, and I will find time to test the Windows version.\n Usage ‌\nI\u0026rsquo;d like to do some examples with sample files, let\u0026rsquo;s prepare the test files:\n# make a test directory mkdir test cd test # chrM.fa as database wget -c https://github.com/quwubin/MFEprimer-3.0/raw/master/chrM.fa # p.fa test primer sequences in fasta format wget -c https://github.com/quwubin/MFEprimer-3.0/raw/master/p.fa # snp in bed format to check whether a primers bind to a snp site wget -c https://github.com/quwubin/MFEprimer-3.0/raw/master/snp.bed # make a softlink ln -s ../mfeprimer-3.1.0-linux-amd64 mfeprimer  The file tree is:\ntree . . ├── chrM.fa ├── mfeprimer -\u0026gt; ../mfeprimer-3.1.0-linux-amd64 ├── p.fa └── snp.bed  Index databases ./mfeprimer index -i chrM.fa  After index, the file tree is:\ntree . . ├── chrM.fa ├── chrM.fa.fai ├── chrM.fa.json ├── chrM.fa.log ├── chrM.fa.primerqc ├── chrM.fa.primerqc.fai ├── mfeprimer -\u0026gt; ../mfeprimer-3.1.0-linux-amd64 ├── p.fa └── snp.bed   mfeprimer index will add 5 files with suffixes: \u0026ldquo;.fai\u0026rdquo;, \u0026ldquo;.json\u0026rdquo;, \u0026ldquo;.primerqc\u0026rdquo;, \u0026ldquo;.primerqc.fai\u0026rdquo; and \u0026ldquo;.log\u0026rdquo;.\n Run mfeprimer for full quality control # run mfeprimer ./mfeprimer -i p.fa -d chrM.fa  Without \u0026ldquo;-o\u0026rdquo; options, the result will print to the screen:\nMFEprimer-3.0 Primer Quality Reports (2019-10-17 11:29:18) Primer ID Sequence (5'--\u0026gt;3') Length GC Tm Dg Binding Number (bp) (%) (°C) (kcal/mol) Plus Minus p1 CTACAACCCCACCACGTACC 20 60.00 60.55 -22.23 0 0 p2 CGTTACACACTTTGCGGCAA 20 50.00 60.47 -22.43 0 0 p3 CTTAAATAGGGACCTGTATGAATGGCTC 28 42.86 61.98 -27.11 2 1 p4 CCGAAATTTTTAATGCAGGTTTGGTAGT 28 35.71 61.97 -27.14 0 1 p5 GGACACTCTATGGGAAAGAGTGTCC 25 52.00 62.91 -26.07 0 1 Hairpin List (1) Hairpin 1: p5 Score: 9, Tm = 26.56 °C, Delta G = -4.48 kcal/mol /////////-------\\\\\\\\\\\\\\\\\\ GGACACTCTATGGGAAAGAGTGTCC Dimer List (1) Dimer 1: p2 x p2 Score: 7, Tm = 14.55 °C, Delta G = -6.09 kcal/mol CGTTACACACTTTGCGGCAA ::::.:::: AACGGCGTTTCACACATTGC Descriptions of [ 1 ] potential amplicons AmpID HitID Size FpTm RpTm FpDg RpDg (bp) (°C) (°C) kcal/mol kcal/mol 1 chrM 200 61.93 62.67 -27.40 -28.08 Amplicon details Amp 1: p3 + p4 ==\u0026gt; chrM Size = 200 bp, GC content = 43.50% F: Tm = 61.93 °C, Delta G = -27.40 kcal/mol R: Tm = 62.67 °C, Delta G = -28.08 kcal/mol Binding sites: 2612(28/28) ... 2811(28/28) \u0026gt;\u0026gt;\u0026gt;p3 1 28 5' CTTAAATAGGGACCTGTATGAATGGCTC 3' :::::::::::::::::::::::::::: 2811 5' CTTAAATAGGGACCTGTATGAATGGCTCcacgagggttcagct...acccacaggtcctaaACTACCAAACCTGCATTAAAAATTTCGG 3' 2612 :::::::::::::::::::::::::::: 3' ACTACCAAACCTGCATTAAAAATTTCGG 5' 28 1 p4\u0026lt;\u0026lt;\u0026lt; \u0026gt;Amp_1 p3 + p4 ==\u0026gt; chrM CTTAAATAGGGACCTGTATGAATGGCTCcacgagggttcagctgtctcttacttttaaccagtgaaattgacctgcccgt gaagaggcgggcataacacagcaagacgagaagaccctatggagctttaatttattaatgcaaacagtacctaacaaacc cacaggtcctaaACTACCAAACCTGCATTAAAAATTTCGG Parameters Primer file: p.fa Database: chrM.fa Kvalue: 9 MisMatch: 0 Tm cutoff (°C): 30.00 Amplicon min size (bp): 0 Amplicon max size (bp): 2000 SNP: Monovalent cations [mM]: 50.00 Divalent cations [mM]: 1.50 dNTPs [mM]: 0.25 Annealing oligos [nM]: 50.00 Cite Kun Wang, Haiwei Li, Yue Xu, Qianzhi Shao, Jianming Yi, Ruichao Wang, Wanshi Cai, Xingyi Hang, Chenggang Zhang, Haoyang Cai, Wubin Qu, MFEprimer-3.0: quality control for PCR primers, Nucleic Acids Research, Volume 47, Issue W1, 02 July 2019, Pages W610–W613, https://doi.org/10.1093/nar/gkz351 Web \u0026amp; Contact https://www.mfeprimer.com Wubin Qu \u0026lt;quwubin@gmail.com\u0026gt; Total time used: 81.641035ms.  The relation between mfeprimer and its sub-commands: mfeprimer do a full quality control for primers for specificity, binding sites, dimers and hairpins. While, spec only for specificity, dimer is only for dimers and hairpin is only for hairpins, as their names indicated. The sub-commands have more options to control the behavious. mfeprimer only have default parameters for dimers and hairpins.\n‌\nRun mfeprimer spec for specificity ./mfeprimer spec -i p.fa -d chrM.fa  ‌\nRun mfeprimer dimer for dimers ./mfeprimer dimer -i p.fa  Run mfeprimer hairpin for hairpins ./mfeprimer hairpin -i p.fa  3. Local web server Download and install  Please download the right version for your machine: https://github.com/quwubin/MFEprimer-3.0/releases (take the Mac version as an example). Uncompress the file. test it.\n# download the executable mfeprimer-web file wget -c https://github.com/quwubin/MFEprimer-3.0/releases/download/v3.1.1/mfeprimer-web-3.1.1-darwin-amd64.tar.gz # uncompress the file tar zxvf mfeprimer-web-3.1.1-darwin-amd64.tar.gz # The file tree is: tree -L 2 . ├── conf │ └── app.conf ├── mfedb │ ├── chrM.fa │ ├── chrM.fa.fai │ ├── chrM.fa.json │ ├── chrM.fa.log │ ├── chrM.fa.primerqc │ └── chrM.fa.primerqc.fai ├── mfeprimer-web ├── static │ ├── css │ ├── fonts │ ├── img │ └── js └── views ├── dimer ├── hairpin ├── layout.html ├── seq └── spec # test it ./mfeprimer-web # open your browser and enter the following address, and you will see the default page. http://localhost:8080/   Configure the server Open the \u0026ldquo;conf/app.conf\u0026rdquo;, change the corresponding item following the references in the file.\nAdd database There is an example database named \u0026ldquo;chrM.fa\u0026rdquo; in \u0026ldquo;mfedb\u0026rdquo; directory. You can place any database here and index the database with command mfeprimer index -i dbname.fasta.\n"});index.add({'id':1,'href':'/publications/','title':"Publications",'content':" Publications  MFEprimer-3.0: Quality Control for PCR Primers. Nucleic Acids Research (2019). https://dx.doi.org/10.1093/nar/gkz351. [NAR, 2019] sRNAPrimerDB: Comprehensive primer design and search web service for small non-coding RNAs[J]. [Bioinformatics, 2018] CRISPR-offinder: a CRISPR guide RNA design and off-target searching tool for user-defined protospacer adjacent motif[J]. [International journal of biological sciences, 2017] Wubin Qu# and Chenggang Zhang. Selecting Specific PCR Primers with MFEprimer. In PCR Primer Design, edited by Chhandak Basu, 1275, 201-213: Springer New York, 2015. [PCR Primer Design, 2nd, 2015] Wubin Qu#, Yang Zhou#, Yanchun Zhang, Yiming Lu, Xiaolei Wang, Dongsheng Zhao, Yi Yang and Chenggang Zhang. MFEprimer-2.0: A fast thermodynamics-based program for checking PCR primer specificity. Nucleic Acids Res. 2012, 40 (W1): W205-W208. [NAR, 2012] Yang Zhou#, Wubin Qu#, Yiming Lu, Yanchuan Zhang, Xiaolei Wang, Dongsheng Zhao, Chenggang Zhang VizPrimer: a web server for visualised PCR primer design based on known gene structure.Bioinformatics, 2011, 27(24):3432-3434. Wubin Qu, Changhong Ren, Yuan Li, Jinping Shi, Jiye Zhang, Xiaolei Wang, Xingyi Hang, Yiming Lu, Dongsheng Zhao and Chenggang Zhang. Reliability analysis of the Ahringer Caenorhabditis elegans RNAi feeding library: a guide for genome-wide screens. BMC Genomics, 2011, 12:170. Zhiyong Shen#, Wubin Qu#, Wen Wang, Yiming Lu, Yonghong Wu, Zhifeng Li, Xingyi Hang, Xiaolei Wang, Dongsheng Zhao, Chenggang Zhang. MPprimer: a program for reliable multiplex PCR primer design.BMC Bioinformatics, 2010, 11:143. (Join first author) Wubin Qu, Zhiyong Shen, Dongsheng Zhao, Yi Yang and Chenggang Zhang. (2009) MFEprimer: multiple factor evaluation of the specificity of PCR primers. Bioinformatics, 25(2), 276-278.  "});index.add({'id':2,'href':'/online-server/','title':"Online-server",'content':" Download and Online Servers Online server  https://mfeprimer3.igenetech.com/ [version 3.0] web server funded by iGeneTech, a company aims to provide kit and service for target capture and sequencing, also is the company the author works for. http://124.207.243.56:8081/ [version 3.1, for test] web server funded by Prof. Dongsheng Zhao (AMMS, 军事医学科学院). Currently, there is only human genome (hg19) available and will add other genomes later.  Download  Download: https://github.com/quwubin/MFEprimer-3.0/releases/ Doc for install and usage  "});index.add({'id':3,'href':'/old-versions/mfeprimer-2.0/','title':"MFEprimer-2.0",'content':" MFEprimer-2.0 Introduction MFEprimer-2.0: A fast thermodynamics-based program for checking PCR primer specificity\nEvaluating the specificity of polymerase chain reaction (PCR) primers is an essential step in PCR primer design. The MFEprimer-2.0 server allows users to check primer specificity against genomic DNA and messenger RNA/complementary DNA sequence databases quickly and easily. MFEprimer-2.0 uses a k-mer index algorithm to accelerate the search process for primer binding sites and uses thermodynamics to evaluate binding stability between each primer and its DNA template. Several important characteristics, such as the sequence, melting temperature and size of each amplicon, either specific or non-specific, are reported on the results page. Based on these characteristics and the user-friendly output, users can readily draw conclusions about the specificity of PCR primers. Analyses for degenerate primers and multiple PCR primers are also supported in MFEprimer-2.0. In addition, the databases supported by MFEprimer-2.0 are comprehensive, and custom databases can also be supported on request. The MFEprimer-2.0 command-line version and local server version are open source and can be downloaded at https://github.com/quwubin/MFEprimer/wiki/Manual/.\nStatus Command-line and local server.\nSource code https://github.com/quwubin/MFEprimer-2.0\n"});index.add({'id':4,'href':'/old-versions/mfeprimer-1.0/','title':"MFEprimer-1.0",'content':" MFEprimer-1.0 Introduction MFEprimer: multiple factor evaluation of the specificity of PCR primers. We developed a program named MFEprimer for evaluating the specificity of PCR primers based on multiple factors, including sequence similarity, stability at the 3’-end of the primer, melting temperature, GC content and number of binding sites between the primer and DNA templates. MFEprimer can help the user to select more suitable primers before running either standard or multiplex PCR reactions. The cDNA and genomic DNA databases of 10 widely used species, as well as user custom databases, were used as DNA templates for analyzing primers specificity.\nStatus Deprecated.\nSource code Not available.\n"});index.add({'id':5,'href':'/categories/','title':"Categories",'content':""});index.add({'id':6,'href':'/','title':"Introduction",'content':" MFEprimer-3.0 Introduction Quality control for primers is crucial for a successful PCR reaction. For example, multiplex PCR is an efficient capture method to simultaneously amplify up to thousands of SNPs in one tube. And it requires compatible primer sets with no non-specific amplifications, no dimers etc. Previous versions of MFEprimer (Qu, et al., 2009, Qu, et al., 2012) mainly focused on primer specificity evaluation. In current version, MFEprimer-3.0 was developed as a full-functional primer quality control program.\nWhat is New?  Binding sites searching is more sensitive by updating the k-mer algorithm to allow mismatches within the k-mer except the first base at the 3’ end. And the binding sites of each primer with stable 3’ end are also listed in output. Implements algorithms to quickly check self-dimers, cross-dimers and hairpins. Command-line version has an option of JSON output to make MFEprimer more useful by acting as a quality control step in “primer design → quality control → redesign” pipeline. Added a function to check whether the binding sites contain SNPs, which will affect the consistency of binding efficiency among different samples. Rewrote all the code by Go language to make the program more reliable, fast and easy to use across multiple platforms.Besides the web server with predefined databases, custom databases upon request and command-line tool for user-defined databases are also supported now.  Usage Download and Online Servers\nHow to cite us  Kun Wang, Haiwei Li, Yue Xu, Qianzhi Shao, Jianming Yi, Ruichao Wang, Wanshi Cai, Xingyi Hang, Chenggang Zhang, Haoyang Cai, Wubin Qu, MFEprimer-3.0: quality control for PCR primers, Nucleic Acids Research, Volume 47, Issue W1, 02 July 2019, Pages W610–W613, https://doi.org/10.1093/nar/gkz351.\n "});index.add({'id':7,'href':'/old-versions/','title':"Old-versions",'content':""});index.add({'id':8,'href':'/tags/','title':"Tags",'content':""});})();